src/main/java/org/mitre/synthea/world/agents/Provider.java

```diff
@@ -743,6 +754,30 @@ public static List<Provider> getProviderList() {
     return new ArrayList<Provider>(providerByUuid.values());
   }
 
+  public static String findProviderStateByNPI(String npi) {
+
+    for (String filename : PROVIDER_SOURCE_FILES) {
+
+      String resource;
+      try {
+        resource = Utilities.readResource(filename, true, true);
+        Iterator<? extends Map<String,String>> csv = SimpleCSV.parseLineByLine(resource);
+  
+        while (csv.hasNext()) {
+          Map<String,String> row = csv.next();
+          String currNpi = row.get("npi");
+
+          if (npi.equals(currNpi) && row.get("state") != null) return Location.getStateNameFromAbbreviation(row.get("state"));
+        }
+      } catch (IOException e) {
+        System.err.println("ERROR: unable to find state for provider by NPI: " + e.getMessage());
+      }
```

Comment:
I don't think this is something you want to catch. You probably should error out of the program if the NPI input is invalid.